src/scripts/settings/webSettings.js

```diff
@@ -1,16 +1,26 @@
 define(['appStorage', 'events'], function (appStorage, events) {
     'use strict';
 
-    var data = {};
+    var data;
+
+    function getConfig() {
+        if (data) {
+            return data;
+        }
 
-    function WebSettings() {
         fetch("/config.json").then(function (response) {
             data = response.json();
         })
+
+        return data;
```

Comment:
this feels wrong... if anything your `getConfig` should be a promise or something...